### ISSUE TITLE
Question on exporting results of intermediate process in metaX

### ISSUE THREAD

**Issue by renzhezhu677** [wenbostar/metaX#6]
Hi again! @wenbostar 
Another question for you : How to export  results of intermediate processes from metaX ?
For my case, which is quite complicated due to a cross-species analysis demand, I need to export the ion intensity after missingvalue filtering and imputation, before normalization. Unfortunately, I couldn't find a function to do this in R document of metaX. 
Thanks!

**Comment by renzhezhu677** [comment]
Well, for exporting middle process results, it seems only possible to run the missing value filtering and imputation step by step instead of using metaXpipe(). Here's my code:
```
p <- filterPeaks(para,ratio=0.2)
p <- filterQCPeaks(para,ratio=0.5)
para <- missingValueImpute(para,method="knn")
```
Actually, the reason I do exporting is that I have samples from 6 species and would like to run missing value filtering and imputation within each species to avoid filtering out features of which the intensity is like a species-specifical pattern. After this, I'd like to merge samples from 6 species into a single table for normalization. Do you think this can work? If this workflow sounds reasonable to you, can I import this merged table into metaX just following the standard pipeline? like this:
```
pfile <- system.file("extdata/MTBLS79.txt",package = "metaX")
rawPeaks(para) <- read.delim(pfile,check.names = FALSE)
```

**Comment by wenbostar** [comment]
Yes, that should work well.

**Comment by renzhezhu677** [comment]
Glad to known it works!
By the way, can you show me the way to export  the middle result?

**Comment by wenbostar** [comment]
You can use function **getPeaksTable**.

**Comment by renzhezhu677** [comment]
It works! Appreciate it!
As I mentioned above, once I merged samples from 6 species into a single table, a new problem arose.：To get the union set of missingvalue-filtered and imputated table of six species will cause new missing values. Since I'd like to choose QC-RSC as the normalization method, and it seems this method can't tolerate any missing value. Therefore, missing value filtering and imputation have to be done to this union set for the second time, in order to do QC-RSC. 

> Actually, the reason I do exporting is that I have samples from 6 species and would like to run missing value filtering and imputation within each species to avoid filtering out features of which the intensity is like a species-specifical pattern. After this, I'd like to merge samples from 6 species into a single table for normalization. 

What worries me is, if I do missing value filtering and imputation to the union set, the species-specific features of 6 species will still be filtered, and what I've done before become meaningless.
One of my rough solution can technically avoid this problem, but I doubt if it's right: do missingvalue filtering except imputation within species. After the merging is done, do missingvalue imputation for the whole union set. Then normalization.
Looking for your suggestions on this problem!

**Comment by wenbostar** [comment]
There are a few different normalization methods in metaX. Some of them can handle the data with missing values.

**Comment by renzhezhu677** [comment]
Hi again! @wenbostar
Recently I saw paper “Lipidomics profiling reveals the role of glycerophospholipid metabolism in psoriasis”.The paper's Data processing mentioned a algorithms：_Prior to statistical analysis, data clean algorithms were applied to the dataset. Features were removed if SNR< 1 (SNR = standard deviationsample/standard deviationQC) or the relative differnce between the mean QC samples intensitycompared to the mean study sample intensity was morethan 3 times the standard deviation of the study sample intensity_. Is this data clean algorithms **pclean** in metaX?
Thanks!
